### PR TITLE
CRUDify Receipt Repository

### PR DESCRIPTION
**Pull request by wingsofovnia**

### CODE REVIEW

File: src/main/java/com/receipts/on/ReceiptRepository.java [hunk 1]
```diff
@@ -30,17 +38,45 @@ public ReceiptRepository(DB db) {
...
+        if (!exists(prescription))
+            return false;
+
+        DBObject query = new BasicDBObject();
+        query.put("prescriptionId", new ObjectId(String.valueOf(prescription.prescriptionId())));
+        collection.update(query, serialize(prescription));
+        return true;
+    }
+
+    public boolean delete(Long id) {
```
Comment (glasierr) [inline src/main/java/com/receipts/on/ReceiptRepository.java]: I don't think that you can actually delete the prescription. What about history?

Comment (YegorB) [reply src/main/java/com/receipts/on/ReceiptRepository.java]: makes sense, but not for MVP

Comment (YegorB) [reply src/main/java/com/receipts/on/ReceiptRepository.java]: In mongodb _id is a String, as for Java driver it is ObjectId, take a look at find

File: src/main/java/com/receipts/on/ReceiptRepository.java [hunk 2]
```diff
@@ -30,17 +38,45 @@ public ReceiptRepository(DB db) {
         return prescriptions;
     }
 
-    public Prescription find(String id) {
+    public Optional<Prescription> find(Long id) {
```
Comment (YegorB) [inline src/main/java/com/receipts/on/ReceiptRepository.java]: String, not Long

**Review by YegorB — APPROVED** [review]

**Pull request merged**